x/amm/keeper/msg_server_exit_pool.go

```diff
@@ -14,7 +14,7 @@ func (k msgServer) ExitPool(goCtx context.Context, msg *types.MsgExitPool) (*typ
 	if err != nil {
 		return nil, err
 	}
-	exitCoins, weightBalanceBonus, slippage, swapFee, _, err := k.Keeper.ExitPool(ctx, sender, msg.PoolId, msg.ShareAmountIn, msg.MinAmountsOut, msg.TokenOutDenom, false, true)
+	exitCoins, weightBalanceBonus, swapFee, slippage, _, err := k.Keeper.ExitPool(ctx, sender, msg.PoolId, msg.ShareAmountIn, msg.MinAmountsOut, msg.TokenOutDenom, false, true)
```

Comment:
could you please add description why are we switching these fields ?

Reply:
> could you please add description why are we switching these fields ?

ExitPool function returns signature is this `(exitCoins sdk.Coins, weightBalanceBonus math.LegacyDec, swapFee math.LegacyDec, slippage math.LegacyDec, takerFeesFinal math.LegacyDec, err error)`

But we were returning `return exitCoins, weightBalanceBonus, slippage, swapFee, takerFeesFinal, nil`

Also, we were expecting the wrong type from the function return signature in the msg_server_exit_pool, so I changed.